pkg/oci/skopeo.go

```diff
@@ -1,45 +1,100 @@
 package oci
 
 import (
-	"os"
-	"os/exec"
+    "context"
+    "fmt"
+    "os"
+    "github.com/containers/image/v5/copy"
+    "github.com/containers/image/v5/signature"
+    "github.com/containers/image/v5/transports/alltransports"
 )
 
 // LoadDocker loads the image to the docker daemon
 func LoadDocker(daemon, dir, imageName string) error {
-	cmd := exec.Command("nix", "run", "nixpkgs#skopeo", "--", "copy", "--insecure-policy", "--dest-daemon-host="+daemon, "dir:"+dir, "docker-daemon:"+imageName)
-	cmd.Stdout = os.Stdout
-	cmd.Stderr = os.Stderr
-	err := cmd.Run()
-	if err != nil {
-		return err
-	}
-
-	return nil
+    ctx := context.Background()
```

Comment:
cant understand the difference between the logic of all these 3 functions, looks pretty much the same to me...